packages/core/src/blocks/baseBlock.ts

```diff
@@ -21,7 +22,7 @@ export type BlockVisitor<T extends object> = (block: BaseBlock, extraData: T) =>
  *
  * It enforces common behavior for all smart filter blocks.
  */
-export abstract class BaseBlock implements ICommandOwner {
+export abstract class BaseBlock implements ICommandOwner, IDisposable {
```

Comment:
I do not think a block should contain anything resource wise ? The runtime and host would, but not the blocks ?

Reply:
Agreed - I'll make this happen at the runtime layer, and add the ability for blocks to register IDisposable objects with the runtime during the generateCommandsAndGatherInitPromises() step.